x-pack/plugins/alerts/server/task_runner/task_runner.ts

```diff
@@ -225,6 +226,14 @@ export class TaskRunner {
       alertInstance.hasScheduledActions()
     );
     const currentAlertInstanceIds = Object.keys(instancesWithScheduledActions);
+
+    scheduleActionsForResolvedInstances(
```

Comment:
I believe this will send a notification for a resolved instance event if the alert instance is muted (or if all instances are muted). Is that the expected behavior?

Reply:
Correct, we shouldn't be scheduling actions if the instance or alert is throttled/muted.  We may want to make that check check a reusable function somewhere, wherever it's done today, so other call sites (like this one), can use it.  Either that, or figure out how to include these new actions to be scheduled in the same place they are done today - sorry I haven't looked at where these places might be yet, but can poke around a bit more if you want - LMK.

Reply:
>Correct, we shouldn't be scheduling actions if the instance or alert is throttled/muted

I believe only `muted` since throttle resets when action group changes.

Reply:
Great catch! By the way, I'm not sure about the reusable function here - we need to identify if the alert is muted to skip the whole `scheduleActionsForResolvedInstances` call and passing down the list of `mutedInstanceIds` to skip it in the schedule function as well. I think we already have this information on the current function level about the specific alert. Or we should get here the most fresh by calling alertsClient get by alertId?

Reply:
I've tested it with the [UI PR](https://github.com/elastic/kibana/pull/83139) part using PagerDuty incidents and it works fine now with handling the `mute` state. Will provide the api integration test as well